src/components/TransactionBlock/PlrDaoStakingTransactionBlock.tsx

```diff
@@ -0,0 +1,632 @@
+      if (active && offers) {
+        setAvailableOffers(offers);
+        setIsLoadingAvailableOffers(false);
+        if (!offers.length) return;
+        const bestOffer: any = offers?.find((offer) => offer.provider == 'Lifi');
+        const selectedOffer = bestOffer?.provider ? bestOffer : offers[0];
+        setSelectedOffer(mapOfferToOption(selectedOffer));
+      }
+    });
+
+    // hook's clean-up function
+    return () => {
+      active = false;
+    };
+  }, [updateAvailableOffers]);
+
+  const availableOffersOptions = useMemo(
+    () => availableOffers?.map(mapOfferToOption),
+    [availableOffers]
+  );
+
+  const receiverAddress = useMemo(() => {
+    if (useCustomAddress) return customReceiverAddress;
+    return selectedReceiveAccountType === AccountTypes.Key
+      ? providerAddress
+      : accountAddress;
+  }, [
+    useCustomAddress,
+    customReceiverAddress,
+    providerAddress,
+    selectedReceiveAccountType,
+    selectedAccountType,
+    accountAddress,
+  ]);
+
+  useEffect(() => {
+    const offer = availableOffers?.find(
+      (availableOffer) => availableOffer.provider === selectedOffer?.value
+    );
+    resetTransactionBlockFieldValidationError(transactionBlockId, 'amount');
+    if (receiverAddress && !isValidEthereumAddress(receiverAddress)) {
+      setTransactionBlockFieldValidationError(
+        transactionBlockId,
+        'receiverAddress',
+        'Invalid receiver address'
+      );
+      return;
+    }
+    resetTransactionBlockFieldValidationError(
+      transactionBlockId,
+      'receiverAddress'
+    );
+    setTransactionBlockValues(transactionBlockId, {
+      hasEnoughPLR,
+      fromChainId: selectedFromNetwork?.chainId ?? undefined,
+      toAsset: hasEnoughPLR ? plrDaoMemberNFT : plrDaoAsset,
+      fromAsset: selectedFromAsset ?? undefined,
+      amount,
+      offer,
+      accountType: selectedAccountType,
+      receiverAddress: receiverAddress ?? undefined,
+    });
+  }, [
+    selectedFromNetwork,
+    selectedFromAsset,
+    amount,
+    selectedOffer,
+    availableOffers,
+    selectedAccountType,
+    receiverAddress,
+  ]);
+
+  const RenderOption = (option: SelectOption) => {
+    const availableOffer = availableOffers?.find(
+      (offer) => offer.provider === option.value
+    );
+    const valueToReceive =
+      availableOffer &&
+      formatAmountDisplay(
+        ethers.utils.formatUnits(
+          availableOffer.receiveAmount,
+          plrDaoAsset.decimals
+        )
+      );
+    return (
+      <OfferDetails>
+        <RoundedImage title={option.title} url={option.iconUrl} size={24} />
+        <div>
+          <Text size={12} marginBottom={2} medium block>
+            {option.title}
+          </Text>
+          {!!valueToReceive && (
+            <Text size={16} medium>
+              {valueToReceive} {plrDaoAsset.symbol}
+            </Text>
+          )}
+        </div>
+      </OfferDetails>
+    );
+  };
+
+  return !isNFTMember ? (
```

Comment:
false negative check, it could be just retuning the bottom code on `isNFTMember` and then after check closes u can return the rest of this below